src/Workleap.DomainEventPropagation.Tests/Subscription/Events/EventsApiIntegrationTests.cs

```diff
@@ -0,0 +1,128 @@
+using System.Net;
+using System.Net.Http.Json;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using Azure.Messaging.EventGrid;
+using Azure.Messaging.EventGrid.SystemEvents;
+using FakeItEasy;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Workleap.DomainEventPropagation.Tests.Subscription.Events;
+
+public class EventsApiIntegrationTests : IClassFixture<EventsApiIntegrationTestsFixture>
+{
+    private readonly HttpClient _httpClient;
+    private readonly ITopicProvider _topicProvider;
+
+    private static readonly JsonSerializerOptions SerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+    };
+
+    public EventsApiIntegrationTests(EventsApiIntegrationTestsFixture fixture)
+    {
+        this._httpClient = fixture.CreateClient();
+        this._topicProvider = fixture.TopicProvider;
+    }
+
+    /// <remarks>
+    /// Unfortunately, <see cref="SubscriptionValidationEventData"/> only has internal
+    /// constructors, which prevents us to use the real object. For tests purpose, it
+    /// is understandable to just a declare a similar type considering their serialized
+    /// representation will be identical.
+    /// </remarks>
+    public class SubscriptionValidationEventTestData
+    {
+        public string ValidationCode { get; init; }
+    }
+
+    [Fact]
+    public async Task GivenEventsApi_WhenASubscriptionEventIsPosted_ThenReturnsOkWithValidationResponse()
+    {
+        A.CallTo(() => this._topicProvider.GetTopicValidationPattern(A<string>._)).Returns("dummytopic");
```

Comment:
We could add some Give When Then comments in the tests 😁

Reply:
# :memo: ✅